boot/assets.go

```diff
@@ -376,20 +398,34 @@ type TrustedAssetsUpdateObserver struct {
 	modeenv *Modeenv
 }
 
-func findMaybeTrustedBootloaderAndAssets(root string, opts *bootloader.Options) (foundBl bootloader.Bootloader, trustedAssets, managedAssets []string, err error) {
-	foundBl, err = bootloader.Find(root, opts)
-	if err != nil {
-		return nil, nil, nil, fmt.Errorf("cannot find bootloader: %v", err)
-	}
-	tbl, ok := foundBl.(bootloader.TrustedAssetsBootloader)
+func trustedAndManagedAssetsOfBootloader(bl bootloader.Bootloader) (trustedAssets, managedAssets []string, err error) {
+	tbl, ok := bl.(bootloader.TrustedAssetsBootloader)
 	if ok {
 		trustedAssets, err = tbl.TrustedAssets()
 		if err != nil {
-			return nil, nil, nil, fmt.Errorf("cannot list %q bootloader trusted assets: %v", foundBl.Name(), err)
+			return nil, nil, fmt.Errorf("cannot list %q bootloader trusted assets: %v", bl.Name(), err)
 		}
 		managedAssets = tbl.ManagedAssets()
 	}
-	return foundBl, trustedAssets, managedAssets, nil
+	return trustedAssets, managedAssets, nil
+}
+
+func findMaybeTrustedBootloaderAndAssets(rootDir string, opts *bootloader.Options) (foundBl bootloader.Bootloader, trustedAssets, managedAssets []string, err error) {
+	foundBl, err = bootloader.Find(rootDir, opts)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("cannot find bootloader: %v", err)
+	}
+	trustedAssets, managedAssets, err = trustedAndManagedAssetsOfBootloader(foundBl)
+	return foundBl, trustedAssets, managedAssets, err
```

Comment:
I don't think managedAssets is used now by any of the callers,  this means it probably shouldn't be returned